scripts/travis/hotrod-integration-test.sh

```diff
@@ -18,14 +18,19 @@ fi
 docker rm -f $CID
 
 BRANCH=${BRANCH:?'missing BRANCH env var'}
-DOCKERHUB_LOGIN=${DOCKERHUB_LOGIN:-false}
 
-# Only push images to Docker Hub for master branch or for release tags vM.N.P and when dockerhub login is done
-if [[ ("$BRANCH" == "master" || $BRANCH =~ ^v[0-9]+\.[0-9]+\.[0-9]+$) && "$DOCKERHUB_LOGIN" == "true" ]]; then
-  echo "upload to Docker Hub, BRANCH=$BRANCH"
+# Only push images to dockerhub/quay.io for master branch or for release tags vM.N.P
+if [[ "$BRANCH" == "master" || $BRANCH =~ ^v[0-9]+\.[0-9]+\.[0-9]+$ ]]; then
+  echo "upload to dockerhub/quay.io, BRANCH=$BRANCH"
 else
-  echo "skip Docker upload, only allowed for tagged releases or master (latest tag)"
+  echo "skip docker images upload, only allowed for tagged releases or master (latest tag)"
   exit 0
 fi
 
-bash ./scripts/travis/upload-to-docker.sh
+DOCKERHUB_USERNAME=${DOCKERHUB_USERNAME:-}
+DOCKERHUB_TOKEN=${DOCKERHUB_TOKEN:-}
+QUAY_USERNAME=${QUAY_USERNAME:-}
+QUAY_TOKEN=${QUAY_TOKEN:-}
+
+bash scripts/travis/upload-to-registry.sh "docker.io" $REPO $DOCKERHUB_USERNAME $DOCKERHUB_TOKEN
+bash scripts/travis/upload-to-registry.sh "quay.io" $REPO $QUAY_USERNAME $QUAY_TOKEN
```

Comment:
```suggestion
bash scripts/travis/upload-to-registry.sh
```

This is the only statement we need here. The enumeration of repositories should be hidden INSIDE `upload-to-registry.sh`.

Reply:
also, passing secrets like this is not safe, they will be printed. But there's no need to pass them, upload-to-registry already has access to env.